rest-api/dao/participant_counts_over_time_service.py

```diff
@@ -176,70 +192,163 @@ def get_enrollment_status_sql(self, filters_sql_ps, filters_sql_p):
     # Add macros for GREATEST and LEAST, as they don't work in SQLite
     # Example: master/rest-api/dao/database_utils.py#L50
 
-    sql = """
-      SELECT
-         SUM(registered_cnt * (cnt_day <= calendar.day)) registered_participants,
-         SUM(member_cnt * (cnt_day <= calendar.day)) member_participants,
-         SUM(full_cnt * (cnt_day <= calendar.day)) full_participants,
-         calendar.day
-       FROM
-       (SELECT c2.day cnt_day,
-               registered.cnt registered_cnt,
-               member.cnt member_cnt,
-               full.cnt full_cnt
-            FROM calendar c2
-            LEFT OUTER JOIN
-            (SELECT COUNT(*) cnt,
-                (CASE WHEN (enrollment_status = 1 OR enrollment_status IS NULL) THEN
-                  DATE(p.sign_up_time)
-                  ELSE NULL END) day
+    sql_by_sample_status_time = """
+      select sum(is_register) as registered_participants,
+       sum(is_member) as member_participants,
+       sum(is_core) as full_participants,
+       z.day
+      from
+      (
+           select
+            x.participant_id,
+            case when register_day<=y.day and member_day>y.day then 1 else 0 end as is_register,
+            case when member_day<=y.day and core_day > y.day then 1 else 0 end as is_member,
+            case when core_day<=y.day then 1 else 0 end as is_core,
+            y.day
+          from
+          (select a.participant_id,a.register_day,COALESCE(b.member_day,'3000-01-01') member_day,COALESCE(b.core_day,'3000-01-01') core_day from
+                (SELECT p.participant_id,
+                       CASE
+                         WHEN (enrollment_status IS NULL or enrollment_status >= 1) THEN DATE(p.sign_up_time)
```

Comment:
I realized that I think this can just be COALESCE(DATE(p.sign_up_time), '3000-01-01') unconditionally... sign_up_time won't be set for anyone who hasn't signed up.

Reply:
Wow, good point.

Reply:
done